grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/AtomicFactory.java

```diff
@@ -54,14 +56,19 @@ public static Atomic create(Atomic atom, ReasonerQuery parent) {
      * @param parent query the created atoms should belong to
      * @return set of atoms
      */
-    public static Set<Atomic> createAtomSet(Conjunction<VarPatternAdmin> pattern, ReasonerQuery parent) {
+    public static Stream<Atomic> createAtoms(Conjunction<VarPatternAdmin> pattern, ReasonerQuery parent) {
         Set<Atomic> atoms = new HashSet<>();
         pattern.varPatterns().stream()
                 .flatMap(var -> var.getProperties()
                         .map(vp -> vp.mapToAtom(var, pattern.varPatterns(), parent))
                         .filter(Objects::nonNull))
                 .forEach(atoms::add);
```

Comment:
Can you not make this whole thing one stream? 
 The `forEach` part here negates some of the benefits of using a stream. If you connected the below stream with this one we could get some performance improvements in the long run. I.e. it will make lazy loading easier.